Disable interactive credential prompt for git resolver

straight-shoota opened the pull request:
Resolves #316

Git repository requests over HTTP can result in 401 Authentication Required status code. This is typically the case when a repository is private. Some git providers also return this status code when a repository exists to avoid disclosing the existence of private repositories.
By default, `git clone` prompts for credentials when it receives a 401 status code. When `shards install` uses `git clone`, this leads to unexpected and misleading behaviour. `shards install` is not meant to be used interactively, so this prompt should be completely disabled.

This PR implements this by using the environment variable `GIT_ASKPASS` set to a value resulting in false. It can still be overriden by setting the environment variable.

bcardiff commented on src/resolvers/git.cr:
```diff
@@ -305,7 +305,7 @@ module Shards
 
     private def mirror_repository
-      run_in_current_folder "git clone --mirror --quiet -- #{Helpers::Path.escape(git_url)} #{local_path}"
+      run_in_current_folder "GIT_ASKPASS=${GIT_ASKPASS:-/usr/bin/test} git clone --mirror --quiet -- #{Helpers::Path.escape(git_url)} #{local_path}"
```
In OSX 

```
$ which test
/bin/test
```

jhass replied:
Does it  need to be an absolute  path or does it do PATH lookup?  Maybe  `${GIT_ASKPASS:-false}` would be most portable.

straight-shoota commented:
The option doesn't need to be an absolute path. I also changed it to `true` because that's the minimal reduced value and should work in any shell (`test` probably too, but we don't really need a test here).
I also changed from using the environment variable `GIT_ASKPASS` to the git config setting `core.askPass` because that seems cleaner. Overriding is still possible by setting the environment variable because it has precedence over the config setting.

jhass commented:
Maybe  add a  comment  explaining this both  :)

I read the diff first and was very confused, as you need to know that the setting takes a command rather than a boolean value and that it's still overridable  :)
